core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -160,6 +160,10 @@ object ByteBufferMessageSet {
             }
         }
       }
+
+      override def close() = {
```

Comment:
Alternatively, instead of closing automatically in case of exceptions, we should never close it and have the callers always call close in a try/finally. Probably easier to reason about.